### DISCUSSION TITLE
Stack navigations are not behaving like a real stack navigation

### DISCUSSION THREAD

**Discussion by champ3oy** [expo/router#731] [Q&A] [answered]
The stack navigation is not working as expected, For example, I have four screens in my app Home, Posts, Post, and Profile.

When I open the Posts screen and clicked on it to open a post on the Post screen, an id is passed to the screen to fetch the corresponding screen. From the Post screen, I click on the author of the post to open the Profile screen, From the author's profile screen I open another post.

**Expectation:** The screens should be stacked in order, so I should have two Post screens in the stack

**What's Happening:** The Post screen that was opened first will be moved to the top of the stack when the post is opened from the profile screen.

Can this be resolved?

**Comment by EvanBacon** [comment, marked as answer]
The default behavior is how stack navigation works, you're describing a different behavior. This can be accomplished in `expo-router@2.0.0-rc.11` by doing something like:

```js
<Screen name="[id]" getId={() => Date.now()}>
```

Creating a random ID will cause the stack to keep adding new screens instead of reusing existing screens.

**Comment by champ3oy** [comment]
Thank you. This helped a alot.
